### ISSUE TITLE
Pylint not detecting type stub modules

### ISSUE THREAD

**Issue by jakeanq** [PyCQA/pylint#10284]
### Bug description

I'm trying to lint using type stubs from the `types-usd` package, which provides type stubs as `pxr-stubs` for the `pxr` USD module.  I would at least like pylint to not complain about missing members.

Example code:
```python
# pylint: disable=missing-docstring,missing-module-docstring
from pxr import Usd


def turtle(name: str) -> Usd.Stage:
    Usd.Stage.turtle()
    return Usd.Stage.Open(name)
```

Possibly related to #10017

### Configuration

```ini

```

### Command used

```shell
pylint --rcfile= test_usd.py
```
Adding `--prefer-stubs y` did not seem to make a difference.

### Pylint output

```
************* Module test_usd
test_usd.py:2:0: E0401: Unable to import 'pxr' (import-error)
```

Note that if `usd-core` is installed (requires Python <3.13 currently), the error becomes:

```
************* Module test_usd
test_usd.py:5:25: E1101: Module 'pxr.Usd' has no 'Stage' member (no-member)
test_usd.py:6:4: E1101: Module 'pxr.Usd' has no 'Stage' member (no-member)
test_usd.py:7:11: E1101: Module 'pxr.Usd' has no 'Stage' member (no-member)
```

### Expected behavior

Ideally a `no-member` error for `Usd.Stage.turtle()`, otherwise no errors would also be reasonable.

Note that `pyright` detects the stubs as expected:
```
.../pylint_usd/test_usd.py
  .../pylint_usd/test_usd.py:6:15 - error: Cannot access attribute "turtle" for class "type[Stage]"
    Attribute "turtle" is unknown (reportAttributeAccessIssue)
1 error, 0 warnings, 0 informations 
```

### Pylint version

```shell
pylint 3.3.5
astroid 3.3.9
Python 3.13.1 (main, Dec  4 2024, 18:05:56) [GCC 14.2.1 20240910]
```
Also tried with Python 3.9.18

### OS / Environment

Python 3.13.1 with Arch Linux
Python 3.9.18 with Rocky 9

### Additional dependencies

```
astroid==3.3.9
dill==0.3.9
isort==6.0.1
mccabe==0.7.0
nodeenv==1.9.1
platformdirs==4.3.6
pylint==3.3.5
pyright==1.1.396
tomlkit==0.13.2
types-usd==24.5.2
typing_extensions==4.12.2
```

**Comment by Pierre-Sassoulas** [comment]
Pylint does not use typing, I'm going to close as a duplicate of #4813
